app/controllers/cards_controller.rb

```diff
@@ -0,0 +1,44 @@
+class CardsController < ApplicationController
+  def index
+    @cards = Card.all
+  end
+
+  def show
+    @card = Card.find(params[:id])
+  end
+
+  def new
+    @card = Card.new
+  end
+
+  def create
+    @card = Card.new(card_params)
+    if @card.save
+      redirect_to @card
+    else
+      render 'new'
+    end
+  end
+
+  def update
+    @card = Card.find(params[:id])
+    if @card.update(card_params)
+      redirect_to @card
+    else
+      render 'edit'
```

Comment:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.